test/unit/package_test.rb

```diff
@@ -1,68 +1,403 @@
+  end
+
+  def test_initialize_increases_dimension_size_to_three
+    @package = Package.new(@weight, [2], @options)
+
+    assert_equal [2, 2, 2], @package.centimetres
+  end
+
+  def test_initialize_default_units
+    assert_equal @dimensions, @package.centimetres
+    assert_equal @weight, @package.grams
+  end
+
+  def test_initialize_units
+    assert_equal @dimensions, @imperial_package.inches
+  end
+
+  def test_initialize_weight_units
+    @package = Package.new(@weight, @dimensions, @options.merge(weight_units: :imperial))
+
+    assert_equal @weight, @package.ounces
+  end
+
+  def test_unpackaged
+    assert Package.new(@weight, @dimensions, unpackaged: true).unpackaged?
+    refute Package.new(@weight, @dimensions, unpackaged: false).unpackaged?
+  end
+
+  def test_oversized
+    assert Package.new(@weight, @dimensions, oversized: true).oversized?
+    refute Package.new(@weight, @dimensions, oversized: false).oversized?
+  end
+
+  def test_gift
+    assert Package.new(@weight, @dimensions, gift: true).gift?
+    refute Package.new(@weight, @dimensions, gift: false).gift?
+  end
+
+  def test_cylinder_tube
+    @package = Package.new(@weight, @dimensions, cylinder: false, tube: false)
+    refute @package.cylinder?
+    refute @package.tube?
+
+    @package = Package.new(@weight, @dimensions, cylinder: true, tube: false)
+    assert @package.cylinder?
+    assert @package.tube?
+
+    @package = Package.new(@weight, @dimensions, cylinder: false, tube: true)
+    assert @package.cylinder?
+    assert @package.tube?
+
+    @package = Package.new(@weight, @dimensions, cylinder: true, tube: true)
+    assert @package.cylinder?
+    assert @package.tube?
+  end
+
+  def test_inches_number_index
+    assert_equal @dimensions[0], @imperial_package.inches(0)
+    assert_equal @dimensions[1], @imperial_package.inches(1)
+    assert_equal @dimensions[2], @imperial_package.inches(2)
+    assert_nil @imperial_package.inches(3)
+  end
+
+  def test_inches_x
+    assert_equal @dimensions[2], @imperial_package.inches(:x)
+    assert_equal @dimensions[2], @imperial_package.inches(:max)
+    assert_equal @dimensions[2], @imperial_package.inches(:length)
+    assert_equal @dimensions[2], @imperial_package.inches(:long)
+  end
+
+  def test_inches_y
+    assert_equal @dimensions[1], @imperial_package.inches(:y)
+    assert_equal @dimensions[1], @imperial_package.inches(:mid)
+    assert_equal @dimensions[1], @imperial_package.inches(:width)
+    assert_equal @dimensions[1], @imperial_package.inches(:wide)
+  end
+
+  def test_inches_z
+    assert_equal @dimensions[0], @imperial_package.inches(:z)
+    assert_equal @dimensions[0], @imperial_package.inches(:min)
+    assert_equal @dimensions[0], @imperial_package.inches(:height)
+    assert_equal @dimensions[0], @imperial_package.inches(:depth)
+    assert_equal @dimensions[0], @imperial_package.inches(:high)
+    assert_equal @dimensions[0], @imperial_package.inches(:deep)
```

Comment:
Just for noting: I'm personally not a fan of this behaviour. It's adding unnecessary complexity / surface area to this library. Are they useful? My own opinion is that they aren't, and that we should let clients decide if they want to alias things.